### DISCUSSION TITLE
Add tooltips to various options

### DISCUSSION THREAD

**Discussion by Tanza3D** [ppy/osu#24234] [Ideas]
To many players who have played this game for a while, most options will seem obvious, but to new players many options, such as "Snaking Sliders", "Interface Voices", "Right mouse drag to absolute scroll", and various others will seem confusing and hard to understand the meaning of, and to many new players, who do not know the game as well as more long-time players, they will not even have a visible effect unless you are paying close attention. I think adding tooltips to most options would be a good idea to help with this.

Obviously, more obvious ones, like "Seasonal Backgrounds," "Background Blur", "Screen mode", and others would not need them, but even to pro players "Threading" might be hard to understand what means without going to Wikipedia, and I think it would be helpful to explain this in-game.

The best way to go about this would be to give tooltips to options options which:
- are not common-place in other games
- do not have an immediate and obvious effect when changing from the main menu
- when changing require a game restart or other long-winded process.

If wanted, I would be happy to PR this myself, though I cannot assure that my wording will be to standard, and it will likely require a lot of modification.

**Comment by bdach** [comment]
These days I generally disagree that adding text is going to be helping anything UX wise. If you want to explain, then show, not tell.

To elaborate:

> Snaking Sliders

This would be eventually moved to the skin editor, and you'd _see_ what the setting does as you toggle it.

> Interface Voices

Could play a voice sample when toggled on. ("welcome to osu!" seems like a good one)

> Right mouse drag to absolute scroll

Would eventually be a skinnable option on the beatmap carousel, and it'd have a drawable hint with a mouse icon with right button pressed showing or something.

> even to pro players "Threading" might be hard to understand what means without going to Wikipedia

You can't explain threading modes in a tooltip. That's what the wiki is for.

Note that as much of the settings menu crap should be moved out to local skin editors, to contextualise the settings to individual components. This means that you at least have an _idea_ which thing you're changing.

If we're desperate then I guess tooltips could be a tolerable stopgap. But I don't know if we're that desperate.
